plugins/catalog-backend-module-gitlab/src/providers/GitlabOrgDiscoveryEntityProvider.ts

```diff
@@ -356,7 +356,22 @@ export class GitlabOrgDiscoveryEntityProvider implements EntityProvider {
     let groups;
     let users;
 
-    if (this.gitLabClient.isSelfManaged()) {
+    // Doesn't matter if we are on SaaS or not, we can either get the users from the defined group or from the root group.
+    if (this.gitLabClient.isSelfManaged() && this.config.restrictUsersToGroup) {
+      groups = (await this.gitLabClient.listDescendantGroups(this.config.group))
```

Comment:
Can we get some tests for this new behaviour? :pray:

Reply:
sure, I added some tests. I hope they are good enough :)